lib/models/device.coffee

```diff
@@ -291,16 +227,9 @@ exports.has = (name, callback) ->
 #		console.log("Is device online? #{isOnline}")
 ###
 exports.isOnline = (name, callback) ->
-
-	if not callback?
-		throw new errors.ResinMissingParameter('callback')
-
-	if not _.isFunction(callback)
-		throw new errors.ResinInvalidParameter('callback', callback, 'not a function')
-
-	exports.get name, (error, device) ->
-		return callback(error) if error?
-		return callback(null, !!device.is_online)
+	exports.get(name).then (device) ->
+		return !!device.is_online
```

Comment:
That should be impossible, it is never allowed to be null - at a guess what happened was you requested for a device that doesn't exist and tried getting the property of an empty object (which returned undefined)